hknweb/exams/views.py

```diff
@@ -12,13 +14,40 @@ def index(request):
 	return render(request, 'exams/index.html', context)
 
 def exams_for_course(request, department, number):
-	department = Department.objects.filter(name__exact=department).get()
+
+	department = Department.objects.filter(abbreviated_name__exact=department).get()
 	course = Course.objects.filter(department__exact=department.id).filter(number__exact=number).get()
 	semesters = CourseSemester.objects.filter(course__exact=course.id)
 
+	print(semesters)
+
 	context = {
 		'course': course,
 		'semesters': semesters
 	}
 
-	return render(request, 'exams/exams-course.html', context)
\ No newline at end of file
+	return render(request, 'exams/exams-course.html', context)
+
+
+def add_exam(request):
+	if request.method == 'POST':
+		form = ExamUploadForm(request.POST, request.FILES)
+		if form.is_valid():
+			semester = form.cleaned_data['semester']
+			course = form.cleaned_data['course']
+			exam = form.cleaned_data['exam']
+			type = form.cleaned_data['type']
+
+			courseSemester = CourseSemester.objects.filter(course=course, semester=semester).first()
+
+			# print(courseSemester)
+			# print(exam + type)
+
+			setattr(courseSemester, exam + ('_sol' if (type == 'sol') else ''), request.FILES['file'])
+			courseSemester.save()
+
+			return HttpResponseRedirect('/')
+		else:
+			print(form.errors)
```

Comment:
I think this error won't actually be visible unless we check the logs. Would be pretty frustrating from the user side if upload fails.

Reply:
Agree with @jameslzhu, if there are errors, let's throw these errors at the very least.

Reply:
From mtg, Do: Return a 400 Status Code
"Tell user Input not formatted / upload failed"

Reply:
Done